### PR TITLE
Channel changed to have capacity

### PR DESCRIPTION
**Pull request by jessberg**
I made channels have capacity as well.  This should lay the groundwork for having both edge and node objects that build off of channels, and then you can optionally set both the nodes' and edges' delays and capacities.  This should make the simulator more general.

### CODE REVIEW

**Comment by fruffy** [comment]
I do not know much about the simulator but the way I understood it was that channels are just layers to links. Channels model additional properties such as delay or queues. Why is capacity needed? Or what role does it serve? Is it bandwidth or maximum queue size? @anirudhSK ?

**Comment by anirudhSK** [comment]
We need to model a limited capacity in RPCs per second for the total egress rate from a node. Because of the limited capacity, there will eventually be queue buildup as well. This is a reasonable model for how network capacity in the public cloud works: https://cloud.google.com/compute/docs/network-bandwidth

**Comment by jessberg** [comment]
> I made channels have capacity as well. This should lay the groundwork for having both edge and node objects that build off of channels, and then you can optionally set both the nodes' and edges' delays and capacities. This should make the simulator more general.

Per the meeting, this has now changed to just give an egress rate to the links.  So the main files changed are sim/src/link.rs and sim/src/main.rs in order to give it an egress rate.  All the other files are changes to Cargo.toml and adding the petgraph modified library so that automated tests run well;  if petgraph is in a separate repo, then the tests can't access it and they fail, so I'm just combining repos.

File: libs/filter_lib/Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ crate-type = ["dylib"]
 path = "src/lib.rs"
 
 [dependencies]
-petgraph = "0.5.1"
+petgraph = { path = "../petgraph" }
```
Comment (fruffy) [inline libs/filter_lib/Cargo.toml:14]: I would recommend adding petgraph as a git submodule and track changes there. Makes maintenance a lot easier.

**Review by fruffy — CHANGES_REQUESTED** [review]

**Review by fruffy — APPROVED** [review]

**Pull request merged**